### DISCUSSION TITLE
Validate the messages that arrive on the NATS server

### DISCUSSION THREAD

**Discussion by remyduthu** [nats-io/nats-server#2350] [Q&A] [answered]
Is it possible to validate the messages that arrive on the NATS server using a JSON Schema for example?

**Comment by ripienaar** [comment]
Not at the moment, we do hav some plans to facilitate that, at present nothing implemented though.

**Comment by derekcollison** [comment, marked as answer]
NATS at its core is payload agnostic. We may look into validation of system level services such as JetStream.

If your clients need messages in a certain form, that logic (at least some) will always need to be at the application layer, or at least outside of NATS core.
